### DISCUSSION TITLE
LIS VIC model with princeton metforcing

### DISCUSSION THREAD

**Discussion by manmeet3591** [NASA-LIS/LISF#842] [Q&A]
This is a follow-up from the thread (https://modelingguru.nasa.gov/thread/5610?tstart=0) on modelling guru. So, I found the VIC parameter file and am trying to run the VIC model with Princeton metforcing. I am getting the following error in the log file 

[INFO] FORCING: Convective Rainfall Rate           1 kg/m2
 [INFO] Using PRINCETON forcing
 [INFO] PRINCETON forcing directory :
 ./INPUT/princeton_v2_2_forcing/

 [INFO] PRINCETON forcing version   :
 2

 [ERR] User-input issue with PRINCETON forcing ...
  -- Currently only supported interpolation options include:
   - bilinear or budget-bilinear -
  Program stopping ...
 [ERR] endrun is being called

The lis config file I am using is as below:

#########################################################################

#Overall driver options
Running mode:                           "retrospective"
Map projection of the LIS domain:       "latlon"
Number of nests:                         1
Number of surface model types:           1 #2
Surface model types:                    "LSM" # "Openwater"
Surface model output interval:          "1hr"
Land surface model:                     "VIC.4.1.1"
#Open water model:                      "template open water"
Number of met forcing sources:          1
Blending method for forcings:           "overlay"
Met forcing sources:                    "PRINCETON"
Topographic correction method (met forcing):    "none"
Enable spatial downscaling of precipitation:    0    0
Spatial interpolation method (met forcing): 'neighbour'  # 'bilinear'  #        "neighbor" 
Spatial upscaling method (met forcing):         "average"
Temporal interpolation method (met forcing):    "linear"

#Runtime options
Forcing variables list file:            ./INPUT/forcing_variables.txt
Output forcing:                         0   #1-yes
Output parameters:                      0   #0- no
Output methodology:                     "2d gridspace"
Output data format:                     netcdf
Output naming style:                    "3 level hierarchy"
Output model restart files:             1
Start mode:                             "coldstart"
Starting year:                          2014
Starting month:                         12
Starting day:                           1
Starting hour:                          0
Starting minute:                        0
Starting second:                        0
Ending year:                            2014
Ending month:                           12
Ending day:                             31
Ending hour:                            23
Ending minute:                          59
Ending second:                          0
Undefined value:                       -9999
Output directory:                      ./OL_OUTPUT
Diagnostic output file:                ./log/lislog_ol
Number of ensembles per tile:            1

#The following options are used for subgrid tiling based on vegetation
Maximum number of surface type tiles per grid:     1
Minimum cutoff percentage (surface type tiles):    0.05
Maximum number of soil texture tiles per grid:     1
Minimum cutoff percentage (soil texture tiles):    0.05
Maximum number of soil fraction tiles per grid:    1
Minimum cutoff percentage (soil fraction tiles):   0.05
Maximum number of elevation bands per grid:        1
Minimum cutoff percentage (elevation bands):       0.05
Maximum number of slope bands per grid:            1
Minimum cutoff percentage (slope bands):           0.05
Maximum number of aspect bands per grid:           1
Minimum cutoff percentage (aspect bands):          0.05

#Processor Layout       
#Should match the total number of processors used
Number of processors along x:    1  # e.g., 32 
Number of processors along y:    1  # e.g., 16 
Halo size along x: 0
Halo size along y: 0

#------------------------ ROUTING -------------------------------------

Routing model:                    "none"

#------------------------RADIATIVE TRANSFER MODELS--------------------------

Radiative transfer model:   "none"

#------------------------APPLICATION MODELS---------------------------------

Number of application models: 0

#---------------------DATA ASSIMILATION ----------------------------------
#Data Assimilation Options

Number of data assimilation instances:               0

Data assimilation algorithm:                         "EnKF"
Data assimilation set:                               "SMOPS soil moisture"
Number of state variables:                           4
Data assimilation use a trained forward model:       0
Data assimilation trained forward model output file: none
Data assimilation exclude analysis increments:       1
Data assimilation output interval for diagnostics:   "1da"
Data assimilation number of observation types:       1
Data assimilation output ensemble members:           0
Data assimilation output processed observations:     0
Data assimilation output innovations:                0

Bias estimation algorithm:                "none"
Bias estimation attributes file:          "none"
Bias estimation restart output frequency:
Bias estimation start mode:
Bias estimation restart file:

#Perturbation options
Perturbations start mode:                 "coldstart"
Perturbations restart output interval:    "1mo"
Perturbations restart filename:           "none"
Apply perturbation bias correction:       0

Forcing perturbation algorithm:           "none" #"GMAO scheme"
Forcing perturbation frequency:           "1hr"
Forcing attributes file:                  ./forcing_attribs.txt
Forcing perturbation attributes file:     ./forcing_pert_attribs.txt

State perturbation algorithm:             "none" #"GMAO scheme"
State perturbation frequency:             "6hr"
State attributes file:                    ./noah_sm_attribs.txt
State perturbation attributes file:       ./noah_sm_pertattribs.txt

Observation perturbation algorithm:       "none" #"GMAO scheme"
Observation perturbation frequency:       "6hr"
Observation attributes file:              ./RTSMOPS_attribs.txt
Observation perturbation attributes file: ./RTSMOPS_pertattribs.txt

# ================================================================

#------------------------DOMAIN SPECIFICATION--------------------------
#Definition of Running Domain
#Specify the domain extremes in latitude and longitude

#Run domain lower left lat:                 -59.5 
#Run domain lower left lon:                -179.5
#Run domain upper right lat:                 89.5
#Run domain upper right lon:                179.5
#Run domain resolution (dx):                  1.0
#Run domain resolution (dy):                  1.0
Run domain lower left lat:                39.005
Run domain lower left lon:              -124.995
Run domain upper right lat:               41.995
Run domain upper right lon:             -122.005
Run domain resolution (dx):         1.0 #       0.01
Run domain resolution (dy):         1.0 #       0.01

##The following options list the choice of parameter maps to be 
##used
#
#The following options list the choice of parameter maps to be used
Landmask data source:            "LDT"
Landcover data source:           "LDT"
Soil texture data source:        "none"
Soil fraction data source:       "none"
Soil color data source:          "none"
Elevation data source:           "none"
Slope data source:               "none"
Aspect data source:              "none"
Curvature data source:           "none"
LAI data source:                 "none"
SAI data source:                 "none"
Albedo data source:              "none"
Max snow albedo data source:     "none"
Greenness data source:           "none"
Roughness data source:           "none"
Porosity data source:            "none"
Ksat data source:                "none"
B parameter data source:         "none"
Quartz data source:              "none"
Emissivity data source:          "none"

LIS domain and parameter data file:  ./lis_input.vic.d01.nc #  ./lis_input.nldas.noah36.d01.nc
Use greenness fraction climatology: 1
Use albedo climatology: 1

#--------------------------------FORCINGS----------------------------------
#NLDAS2 (suppforcing option =5)
#NLDAS2 forcing directory:               ./INPUT/NLDAS2.FORCING
#NLDAS2 data center source:              "GES-DISC"
##NLDAS2 domain x-dimension size:         27 #28
##NLDAS2 domain y-dimension size:         21 #20
#NLDAS2 use model level data:            0
#NLDAS2 use model based swdown:          0
#NLDAS2 use model based precip:          0
#NLDAS2 use model based pressure:        0
#--------------------------------FORCINGS----------------------------------
PRINCETON forcing directory:     ./INPUT/princeton_v2_2_forcing/ # ./input/MET_FORCING/PRINCETON
PRINCETON forcing version: "2"
#Enable spatial downscaling of precipitation: 1
#Spatial interpolation method (met forcing): bilinear
#------------------------OPEN WATER MODELS---------------------------
#
#Template open water timestep:             "15mn"

#-----------------------LAND SURFACE MODELS--------------------------
# == VIC 4.1.1 Parameter Files ==
VIC411 vegetation param file: ./INPUT/LS_PARAMETERS/vic_parms/vegparam_1.0
VIC411 soil param file:       ./INPUT/LS_PARAMETERS/vic_parms/soilparam_1.0
VIC411 number of soil cols:      53
VIC411 number of root zones:     2
VIC411 global LAI flag:          1       # 0: LAI from veg library; 1: LAI from veg param file
VIC411 spatial transform:        none    # Leave as 'none' for now
VIC411 map projection:          latlon
#VIC411 lower left lat:         25.0625
#VIC411 lower left lon:       -124.9375
#VIC411 upper right lat:        52.9375
#VIC411 upper right lon:       -67.0625
#VIC411 resolution (dx):         1.0
#VIC411 resolution (dy):         1.0
VIC411 lower left lat:                     39.005
VIC411 lower left lon:                   -124.995
VIC411 upper right lat:                    41.995
VIC411 upper right lon:                  -122.005
VIC411 resolution (dx):       1.0 #              0.01
VIC411 resolution (dy):       1.0 #              0.01

VIC411 model timestep:             1hr  # VIC model-step
VIC411 restart output interval:    1da
VIC411 veg tiling scheme:          1      # 0=VIC tiling; 1=LIS tiling
VIC411 global parameter file:     ./vic_parameters.txt # ./vic411_global_file_nldas2_testcase 
VIC411 total number of veg types:  13
VIC411 convert units:              1
Model output attributes file:  ./INPUT/NOAH36_OUTPUT_LIST.TBL

Output start year:
Output start month:
Output start day:
Output start hour:
Output start minutes:
Output start seconds:

####################################################################

**Comment by karsenau** [comment]
Hi @manmeet3591,

I see that you are trying to run VIC with Princeton forcing option in LIS.  However, based on the error you are receiving and what you shared from your lis.config file, it looks as though the Princeton forcing reader doesn't currently support "neighbor" interpolation option:

Spatial interpolation method (met forcing): 'neighbour' # 'bilinear' # "neighbor"

Your LIS runtime error states:

[ERR] User-input issue with PRINCETON forcing ...
-- Currently only supported interpolation options include:

**bilinear or budget-bilinear -** <<==  
Program stopping ...
[ERR] endrun is being called

So depending on your run domain, you could select "budget-bilinear" for now.  If you are running at almost the same resolution as the Princeton forcing (e.g., 1.0 deg), you could submit a GitHub LISF issue "ticket" under our "Issues" tab to request to add the "neighbor" option.

Also, make sure to enter "neighbor" in the American English spelling and not British version of "neighbour".  (Perhaps someday we can support both spellings).

Best regards,
karsenau

**Reply by manmeet3591** [reply]
Unfortunately this is this is also throwing up the same error.

**Reply by manmeet3591** [reply]
I am using the same lis config with the option 
Spatial interpolation method (met forcing): 'bilinear'
I have also tried the budget-bilinear method

**Comment by karsenau** [comment]
Hi @manmeet3591,

Thank you for reporting that. I went back to look again at your original post on ModelingGuru:
https://modelingguru.nasa.gov/thread/5610?tstart=0

And I wanted to confirm if this is the LIS run domain that you are trying to run on:

Run domain lower left lat:          39.0625
Run domain lower left lon:        -124.9375
Run domain upper right lat:         41.9375
Run domain upper right lon:       -122.0625
Run domain resolution (dx):          1.0
Run domain resolution (dy):          1.0

If so, I would recommend a few things:
1) For a latlon run grid, make sure to have the gridcell midpoints of your grid's lat/lon extents reflect that of the spatial resolution that you are running on. For example, with a 1.0 x 1.0 deg spatial resolution, we would recommend using 0.5 for the extent gridcell midpoints, such as:
Run domain lower left lat:          39.5
Run domain lower left lon:        -124.5
Run domain upper right lat:         41.5
Run domain upper right lon:       -122.5

LIS run domains use grids that are representative of the center point of a gridcell (and not a corner point of a gridcell).

2) Next, many of the parameter files that you are working with for VIC.4.1.1, you are reading in are at their "native" resolutions, meaning that you are reading in 0.25 deg resolution soils files and 0.01 deg resolution vegetation files, without performing any spatial transforms (e.g., average, mode, tile), on the files to the 1.0 deg resolution that you are trying to run at.

We have supported more the version of VIC.4.1.2, which we have more documentation support for. 

Also, I believe you will need to run with specific VIC-supported parameter files.  We can check with @shugongwang, who helped implement the original VIC versions, to see if he can provide additional insight and information about which parameter files need to be read in to support running VIC v.4.1.1 or v.4.1.2.

Thanks again!

**Comment by manmeet3591** [comment]
Hi @karsenau thanks for your reply. 

1. So as you suggested, I have tried with the following settings 
Run domain lower left lat: 39.5
Run domain lower left lon: -124.5
Run domain upper right lat: 41.5
Run domain upper right lon: -122.5
But still getting the same error
 ./INPUT/princeton_v2_2_forcing/

 [INFO] PRINCETON forcing version   :
 2

 [ERR] User-input issue with PRINCETON forcing ...
  -- Currently only supported interpolation options include:
   - bilinear or budget-bilinear -
  Program stopping ...
 [ERR] endrun is being called

2. I have tried using the tile spatial transform for landcover as specified in https://github.com/NASA-LIS/LISF/blob/master/lis/testcases/surfacemodels/land/vic.4.1.1/ldt.config

Should I try with VIC.4.1.2 ?

I was using the vic_parameters file available openly. I am sharing the ldt.config, lis.config, lis_input.nc and vic_parameters.txt at https://drive.google.com/drive/folders/16cusB-NtTYNm9Y-Gj2ci4kItIJ-kllo8?usp=sharing @shugongwang @karsenau 

Thank you so much for the support. Looking forward to a solution with running setup.

**Reply by shugongwang** [reply]
Apart from your interpolation error, what is the original resolution of your VIC parameter file? I noticed that you use 1 degree in your LDT configuration for the domain resolution. If my memory servers, LDT cannot interpolate VIC parameters. This means that your original VIC parameters should be at 1 degree. We can solve your interpolation error after we get this question cleared.

**Reply by manmeet3591** [reply]
@shugongwang I am unsure which parameter file are you talking about. If it is vic_paramaters.txt I was not able to find the resolution in that file. I am sorry if this response is too naive as I am new to both VIC and LIS.

**Reply by manmeet3591** [reply]
The vic_parameters.txt that I am using is available in the same google drive link as I shared before. The specific file can also be accessed from https://drive.google.com/drive/u/4/folders/16cusB-NtTYNm9Y-Gj2ci4kItIJ-kllo8

**Reply by shugongwang** [reply]
I don't have access to your google drive. But if the following two parameter files are at 1.0 degree, you should be fine.
VIC411 vegetation param file: ./INPUT/LS_PARAMETERS/vic_parms/vegparam_1.0
VIC411 soil param file: ./INPUT/LS_PARAMETERS/vic_parms/soilparam_1.0

**Reply by manmeet3591** [reply]
@shugongwang I am very sorry about that. I have now changed the access settings (https://drive.google.com/drive/folders/16cusB-NtTYNm9Y-Gj2ci4kItIJ-kllo8?usp=sharing). Also sharing the vic_parameters.txt file below
```
###############################################################################
 # @section DESCRIPTION
 #
 # VIC Model Parameters Namelist
 #
 # @section LICENSE
 #
 # The Variable Infiltration Capacity (VIC) macroscale hydrological model
 # Copyright (C) 2014 The Land Surface Hydrology Group, Department of Civil
 # and Environmental Engineering, University of Washington.
 #
 # The VIC model is free software; you can redistribute it and/or
 # modify it under the terms of the GNU General Public License
 # as published by the Free Software Foundation; either version 2
 # of the License, or (at your option) any later version.
 #
 # This program is distributed in the hope that it will be useful,
 # but WITHOUT ANY WARRANTY; without even the implied warranty of
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU General Public License for more details.
 #
 # You should have received a copy of the GNU General Public License along with
 # this program; if not, write to the Free Software Foundation, Inc.,
 # 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 ##############################################################################

###############################################################################
LAPSE_RATE -0.0065   # temperature lapse rate in C/m
###############################################################################

###############################################################################
ALBEDO_BARE_SOIL 0.2  # albedo for bare soil

EMISS_GRND 0.97  # Ground emissivity
EMISS_VEG 0.97  # Veg emissivity
EMISS_ICE 0.97  # Ice emissivity
EMISS_SNOW 0.97  # Snow emissivity
EMISS_H2O 0.98 # Emissivity of open water surface
###############################################################################

###############################################################################
# Physical Constraints
SOIL_RESID_MOIST 0.0  # define residual moisture content of soil column
SOIL_SLAB_MOIST_FRACT 1.0  # Volumetric moisture content (fraction of porosity)
                           # in the soil/rock below the bottom soil layer; this
                           # assumes that the soil below the bottom layer has
                           # the same texture as the bottom layer.

# water holding capacity of snow as a fraction of snow-water-equivalent
VEG_LIQUID_WATER_CAPACITY 0.035

# multiplier to calculate the amount of available snow interception as a
#function of LAI (m)
VEG_LAI_SNOW_MULTIPLIER 0.0005

# the amount of snow on the canopy that can only be melted off. (m)
VEG_MIN_INTERCEPTION_STORAGE 0.005

# Coefficient multiplied by the LAI to determine the amount of water that can
# be stored in the canopy
VEG_LAI_WATER_FACTOR 0.1
###############################################################################

###############################################################################
# Canopy Resistance Parameters
CANOPY_CLOSURE 4000.0  # Threshold vapor pressure deficit for stomatal closure (Pa)
CANOPY_RSMAX 5000.0  # Maximum allowable resistance (s/m)
CANOPY_VPDMINFACTOR 0.1  # Minimum allowable vapor pressure deficit factor
###############################################################################

###############################################################################
# Lake Model Parameters

# lake model related constants
LAKE_TMELT 0.0
LAKE_MAX_SURFACE_LAKE 0.6  # max. surface layer thickness for E-B (m)
LAKE_BETA 0.001  # Curve shape parameter for lake profile.
LAKE_FRACMIN 0.10  # min ice thickness in meters
LAKE_FRACLIM 0.02  # lower limit on fractional ice cover
LAKE_DM 1.38889E-07  # molecular diffusivity of water
LAKE_SNOWCRIT 0.05  # for albedo, in m
LAKE_ZWATER 0.0045  # 0.004 - original value
LAKE_ZSNOW 0.005
LAKE_RHOSNOW 250.0  # densities snow on ice
LAKE_CONDI 2.3  # thermal conductivity of ice
LAKE_CONDS 0.7  # thermal conductivity of snow

# attenuation of short and longwave radiation through ice (1/m)
LAKE_LAMISW 1.5  # 1.5 in Patterson & Hamblin
LAKE_LAMILW 20.0  # 20.0 in Patterson & Hamblin

# attenuation of short and longwave radiation through snow (1/m)
LAKE_LAMSSW 6.0  # 6.0 in Patterson & Hamblin
LAKE_LAMSLW 20.0  # 20.0 in Patterson & Hamblin

# attenuation of short and longwave radiation through water (1/m)
LAKE_LAMWSW 0.3  # San Fran Bay data: 0.31 - 29.9 1/m (visible)
LAKE_LAMWLW 1.4  # Hostetler and Bartlein assume 0.85 1/m (total)
LAKE_A1 0.7  # Fraction of radiation in visible band.
LAKE_A2 0.3  # Fraction of radiation in infrared band.
LAKE_QWTAU 43200.0  # D. Pollard sub-ice time constant. (86400. / 2.)
###############################################################################

###############################################################################
# Saturated Vapor Pressure Curve Parameters
SVP_A 0.61078
SVP_B 17.269
SVP_C 237.3
###############################################################################

###############################################################################
# Carbon Cycling constants
CARBON_CATMCURRENT 383.0  # Current global atmospheric CO2 mixing ratio (ppm)
CARBON_SW2PAR 0.45  # Empirical ratio of PAR [W/m2] to SHORTWAVE [W/m2] from Lopez et al., 2001
###############################################################################

###############################################################################
# Photosynthesis Parameters
PHOTO_OMEGA 0.12  # single leaf scattering albedo
PHOTO_LAIMAX 8.0  # Maximum LAI in nitrogen scaling
PHOTO_LAILIMIT 3.0  # Minimum LAI in nitrogen scaling and maximum LAI in PAR
                    # computation
PHOTO_LAIMIN 1.0e-9  # Minimum LAI in PAR computation
PHOTO_EPAR 2.2e5  # Energy content of PAR [J/mol photons]
                    # = (4.6 mol/MJ PAR)^-1
PHOTO_FCMAX 0.9  # Maximum fractional veg cover
                   # (1-FcMax) = min amount of ground visible
PHOTO_FCMIN 1.0e-3  # Minimum fractional veg cover
                      # (1-FcMin) = max amount of ground visible
PHOTO_ZENITHMIN 0.0174524  # Check for solar zenith angle > 89 deg
PHOTO_ZENITHMINPAR 1.0e-3  # Cosine of the minimum solar zenith angle for
                            # photosynthesis to take place
PHOTO_ALBSOIPARMIN 0.0  # Minimum soil reflectivity in PAR range
PHOTO_MINMAXETRANS 1e-12  # Minimum of maximum electron transport rate
                          # [10e-12 mol/(m^2 s)]
PHOTO_MINSTOMCOND 0.0  # Minimum stomatal conductance [mol H2O/m2s]

# Factors that relate leaf internal CO2 concentration to ambient CO2
# concentration
PHOTO_FCI1C3 0.87  # C3 Plants
PHOTO_FCI1C4 0.67  # C4 Plants

# C3 PLANTS: FARQUHAR, G.D., S. VON CAEMMERER AND J.A. BERRY, 1980. A
# BIOCHEMICAL MODEL OF PHOTOYNTHESIS IN LEAVES OF C3 SPECIES. PLANTA 149,
# 78-90.
PHOTO_OX 0.21  # OXYGEN CONCENTRATION [MOL(O2) / MOL(AIR)]
PHOTO_KC0 460.0e-6  # MICHAELIS-MENTEN CONSTANT FOR CO2 AT 25C
                   #  [MOL(CO2) / MOL(AIR)]
PHOTO_KO0 330.0e-3  # MICHAELIS-MENTEN CONSTANT FOR O2 AT 25C
                   # [MOL(O2) / MOL(AIR)]
PHOTO_EC 59356.0  # ACTIVATION ENERGY FOR KC [J / MOL]
PHOTO_EO 35948.0  # ACTIVATION ENERGY FOR KO [J / MOL]
PHOTO_EV 58520.0  # ACTIVATION ENERGY FOR VCMAX [J / MOL]
PHOTO_ER 45000.0  # ACTIVATION ENERGY FOR DARK RESPIRATION [J / MOL]
PHOTO_ALC3 0.28    # EFFICIENCY OF OF PHOTON CAPTURE
PHOTO_FRDC3 0.011  # RATIO OF DARK RESPIRATION TO "PVM" AT 25C for C3

# C4 PLANTS: COLLATZ, G.J., M. RIBAS-CARBO AND J.A. BERRY, 1992.  COUPLED
# PHOTOSYNTHESIS-STOMATAL CONDUCTANCE MODEL FOR LEAVES OF C4 PLANTS. AUST. J.
# PLANT PHYSIOL. 19, 519-538.
PHOTO_EK 50967.0  # Q10=2 (Collatz et al. 1992)
PHOTO_ALC4 0.04  # EFFECTIVE QUANTUM EFFICIENCY
PHOTO_FRDC4 0.042  # RATIO OF DARK RESPIRATION TO "PVM" AT 25C for C4
PHOTO_THETA 0.83  # CURVATURE PARAMETER

# Plant Maintenance and Growth Respiration Parameters
PHOTO_FRLEAF 0.4  # Ratio of canopy leaf respiration to whole plant
                  # maintenance respiration
PHOTO_FRGROWTH 0.25  # Ratio of plant growth respiration to NPP
###############################################################################

###############################################################################
# Soil Respiration Parameters
SRESP_E0_LT 308.56  # Lloyd-Taylor E0 parameter [K]
SRESP_T0_LT 227.13  # Lloyd-Taylor T0 parameter [K]
SRESP_WMINFM 0.0  # minimum soil moisture (fraction) at which soil
                    # respiration can occur
SRESP_WMAXFM 1.0  # maximum soil moisture (fraction) at which soil
                    # respiration can occur
SRESP_WOPTFM 0.5  # soil moisture (fraction) at which maximum soil
                    # respiration occurs
SRESP_RHSAT 0.15  # ratio of soil respiration rate under saturated conditions
                    # (w=wmaxFM) to that under optimal conditions (w=woptFM)
SRESP_RFACTOR 0.5  # scaling factor to account for other (non-moisture)
                     # sources of inhibition of respiration
SRESP_TAULITTER 2.86  # Litter pool turnover time [y]
SRESP_TAUINTER 33.3  # Intermediate pool turnover time [y]
SRESP_TAUSLOW 1000.0  # Slow pool turnover time [y]
SRESP_FAIR 0.7  # Fraction of respired carbon from litter pool that is lost
                  # to atmosphere
SRESP_FINTER 0.985  # Fraction of [respired carbon from litter pool that goes
                      # to soil] that goes to intermediate pool
###############################################################################

###############################################################################
# Iteration bracket widths
SNOW_DT 5.0  # Used to bracket snow surface temperatures while computing the
               # snow surface energy balance (C)
SURF_DT 1.0  # Used to bracket soil surface temperatures while computing
               # energy balance (C)
SOIL_DT 0.25  # Used to bracket soil temperatures while solving the soil
                # thermal flux (C)
CANOPY_DT 1.0  # Used to bracket canopy air temperatures while computing
                 #energy balance (C)
CANOPY_VP 25.0  # Used to bracket canopy vapor pressures while computing
                  # moisture balance (Pa)
###############################################################################

###############################################################################
# Snow Parameters

# maximum depth of the surface layer in water equivalent (m) [default 0.125]
SNOW_MAX_SURFACE_SWE 0.125

# density of new fallen snow [50]
SNOW_NEW_SNOW_DENSITY 50.0

# Density limit used in calculation of destructive metamorphism
SNOW_DENS_DMLIMIT 100.0  # (kg/m^3)

SNOW_DENS_MAX_CHANGE 0.9

# Constants in snow density computation
SNOW_DENS_ETA0 3.6e6  # viscosity of snow at T=0C and density = 0
                        # used in calculation of true viscosity (Ns/m2)
SNOW_DENS_C1 0.04
SNOW_DENS_C2 2.778e-6
SNOW_DENS_C5 0.08  # constant used in snow viscosity calculation, taken from
                     # SNTHRM.89 (/C)
SNOW_DENS_C6 0.021  # constant used in snow viscosity calculation, taken from
                      # SNTHRM.89 (kg/m3)
SNOW_DENS_F 0.6  # internal compaction rate coefficient

# Minimum SWQ for which the snowpack energy balance is computed independent of
# the soil surface temperature
SNOW_MIN_SWQ_EB_THRES 0.0010

# Attenuation coefficients for shortwave in a snowpack.  Values and equation
# taken from Patterson and Hamblin, 1988
SNOW_A1 0.7
SNOW_A2 0.3
SNOW_L1 6.0  # (1/m)
SNOW_L2 20.0  # (1/m)

# Snow albedo curve parameters.  Defaults are from Bras p263. Should not be
# changed except for serious problems with snow melt
SNOW_NEW_SNOW_ALB 0.85
SNOW_ALB_ACCUM_A 0.94
SNOW_ALB_ACCUM_B 0.58
SNOW_ALB_THAW_A 0.82
SNOW_ALB_THAW_B 0.46

# Defines the minimum amount of new snow (mm) which will reset the snowpack
# albedo to new snow
SNOW_TRACESNOW 0.03
###############################################################################

###############################################################################
# Blowing Snow Parameters
BLOWING_KA 0.0245187  # thermal conductivity of air (W/mK)
BLOWING_CSALT 0.68  # saltation constant m/s
BLOWING_UTHRESH 0.25  # threshold shear velocity m/s
BLOWING_KIN_VIS 1.3e-5  # Kinemativ viscosity of air (m2/s)
BLOWING_MAX_ITER 100  # Max. iterations for numerical integration
BLOWING_K 5
BLOWING_MACHEPS 1.0e-6  # Accuracy tolerance for numerical integration
BLOWING_SETTLING 0.3  # Particle settling velocity m/s
BLOWING_NUMINCS 10  # Number of prob intervals to solve for wind.
###############################################################################

###############################################################################
# July average temperature for tree line
TREELINE_TEMPERATURE 10.0
###############################################################################
```

**Reply by shugongwang** [reply]
Can you please share the following to files too?
VIC411 vegetation param file: ./INPUT/LS_PARAMETERS/vic_parms/vegparam_1.0
VIC411 soil param file: ./INPUT/LS_PARAMETERS/vic_parms/soilparam_1.0

**Reply by manmeet3591** [reply]
Apologies for the delay, please find the files vegparam_1.0 and soilparam_1.0
https://drive.google.com/file/d/1Lfb2Oztx1SfMlIKjW-5KecKc78kGBg97/view
https://drive.google.com/file/d/1847OadNHzldCVunWxY1EdQOXlzJCouPf/view

**Comment by karsenau** [comment]
Thanks, @shugongwang, for your reply and help with the parameter side. Also, do you recommend VIC4.1.2 over VIC4.1.1 in LIS? Or does it not matter.

Hi @manmeet3591,

In the error you just reported (in your last post to us) with the Princeton version "2" forcing, was this error being encountered in LDT?  For example, you copied from your log file the following:

[INFO] PRINCETON forcing version :
2

[ERR] User-input issue with PRINCETON forcing ...
-- Currently only supported interpolation options include:

bilinear or budget-bilinear -
Program stopping ...
[ERR] endrun is being called

If this is from your LDT parameter processing run, I noticed that you have the following set in your ldt.config file:

Met forcing sources:                    "PRINCETON"
Number of met forcing sources:           0                # Enter number of forcing types
Met forcing sources:                        "none"             # Enter 'none' if no forcing selected
Met spatial transform methods:        none              # bilinear | budget-bilinear | neighbor | average
Topographic correction method (met forcing):  "none"    # none | lapse-rate

If you are not turning on any forcing data (e.g., Princeton) elevation / terrain file to be read in, you will want to set "PRINCETON" here to "none". That should fix one problem in your LDT config file.

Also, in relation to the Princeton version 2 forcing in LIS, since you set your run domain to the same resolution as the Princeton forcing, which is 1.0 x 1.0 deg too, you should be able to run with "bilinear" as the "interpolation" option in LIS, with VIC or any other LSM. Did you encounter any issues when running LIS with Noah LSM at 1.0 x 1.0 deg and Princeton forcing?

Here are the Princeton forcing dimensions:
       if (princeton_struc(n)%version == "2" .OR. princeton_struc(n)%version == "2.2") then
          gridDesci(4) = -89.50
          gridDesci(5) = -179.50
          gridDesci(6) = 128
          gridDesci(7) = 89.50
          gridDesci(8) = 179.50
          gridDesci(9) = 1.00
          gridDesci(10) = 1.00

Thanks!

**Reply by manmeet3591** [reply]
@karsenau I am not getting this issue while running LDT, this is when I am trying to run LIS. Also, I did not encounter any such issues when running LIS with Noah LSM at 1.0 x 1.0 deg and Princeton forcing. I am trying to modify the same options that worked for Noah LSM with Princeton forcing.

**Reply by karsenau** [reply]
Hi @manmeet3591,  
Thank you for confirming that information. This could be possibly a landmask mismatch with the landcover for the VIC LSM. Sometimes, LDT may produce an input file for LIS, but parameters may not be correct.

Would you mind posting or attaching your LDT log file?

**Reply by manmeet3591** [reply]
Hi @karsenau sure. You may please find the ldt log file at https://drive.google.com/file/d/1NtZ10IP9o9z3u4VYkvoqVS4vrEGk92Vq/view?usp=sharing

**Reply by karsenau** [reply]
Hi @manmeet3591,
Thank you for sharing your ldtlog.0000 file. I couldn't see anything obvious that could be wrong from your log file, and hopefully it is just simply related to which and how the VIC parameters themselves are read in.

One other follow-up question, are you also trying to apply any "lapse-rate" topographic correction in LIS to the Princeton forcing when running VIC (or Noah)?

**Reply by manmeet3591** [reply]
Hi @karsenau Thanks for your response. No I am not trying any lapse-rate topographic correction in LIS to the Princeton forcing when running VIC or Noah

**Reply by karsenau** [reply]
Ok, then it may be the VIC parameters.

**Comment by manmeet3591** [comment]
@karsenau @shugongwang @sujayvkumar can you please help me with this issue

**Reply by shugongwang** [reply]
How about try other forcing data, such as NLDAS which has been extensively tested with VIC? If you still have problems, then the problems could be related to the model parameters/configurations. Otherwise, the problems should be related to forcing setup.
